src/create_disk.sh

```diff
@@ -0,0 +1,77 @@
+#!/bin/sh
+
+export PATH=$PATH:/sbin:/usr/sbin
+
+usage() {
+	echo 'create_disk <device> <ostree-repo> <ostree-ref> <grub-script>'
+	exit 1
+}
+
+disk="$1"
+shift || usage
+ostree="$1"
+shift || usage
+ref="$1"
+shift || usage
+grub_script="$1"
+shift || usage
+
+set -e
+
+# partition and create fs
+sgdisk -Z $disk \
+	-n 1:0:+128M -c 1:boot \
+	-n 2:0:+128M -c 2:EFI-SYSTEM -t 2:C12A7328-F81F-11D2-BA4B-00A0C93EC93B \
+	-n 3:0:+128M -c 3:BIOS-BOOT  -t 3:21686148-6449-6E6F-744E-656564454649 \
+	-n 4:0:0     -c 4:root       -t 4:4F68BCE3-E8CD-4DB1-96E7-FBCAF984B709
+sgdisk -p "$disk"
+
+# HACK ALERT - wait for partition rescans
+sleep 2
+
+mkfs.ext4 "${disk}1" -L boot
+mkfs.fat "${disk}2" -n EFI-SYSTEM
+# partition 3 has no FS, its for bios grub
+mkfs.xfs "${disk}4"  -L root
+
+# mount the partitions
+rm -rf rootfs
+mkdir rootfs
+mount "${disk}4" rootfs
+mkdir rootfs/boot
+mount "${disk}1" rootfs/boot
+mkdir rootfs/boot/efi
+mount "${disk}2" rootfs/boot/efi
+
+# init the ostree
+ostree admin init-fs rootfs
+ostree pull-local "$ostree" --repo rootfs/ostree/repo
+ostree admin os-init fedora-coreos --sysroot rootfs
+kargs='root=/dev/disk/by-label/root console=ttyS0,115200n8 console=ttyS0 rootflags=defaults,prjquota rw $ignition_firstboot'
```

Comment:
Can we move the middle part of this to two variables (`$consoleflags` and `$rootflags`) that can be overridden via environment?

Reply:
Also we need to use `VM_TERMINAL` instead of `ttyS0`.

Reply:
@lucab what environment precisely? This is run in supermin, so the environment is pretty sparse.

The kargs currently live in a kickstart. Seems like they ought to live in `image.yaml`?

Reply:
https://github.com/coreos/coreos-assembler/pull/525

Reply:
Summarizing some discussion on `#fedora-coreos`: We're going to keep the kargs that relate to this script in this script but move the other kargs to cmd-build since we can do thing like templating (for things like `$VM_TERMINAL` there since it's not in supermin. We may move them later, but that's for another PR.